protected_branches.go

```diff
@@ -138,6 +138,7 @@ type ProtectRepositoryBranchesOptions struct {
 type BranchPermissionOptions struct {
 	UserID      *int              `url:"user_id,omitempty" json:"user_id,omitempty"`
 	GroupID     *int              `url:"group_id,omitempty" json:"group_id,omitempty"`
+	DeployKeyId *int              `url:"deploy_key_id,omitempty" json:"deploy_key_id,omitempty"`
```

Comment:
Can you show me where in the docs or the code this field is mentioned? And for it to be a proper Go field name, it should be `DeployKeyID`

Reply:
Hi @svanharmelen,
I renamed the field.
The feature is documented [here](https://docs.gitlab.com/ee/user/project/protected_branches.html#allow-deploy-keys-to-push-to-a-protected-branch). They however have failed to update the API documentation when adding the feature. [Here](https://gitlab.com/gitlab-org/gitlab/-/issues/30769) is a link to the issue that introduced the change with all the related work.